public/app/core/services/echo/EchoSrv.ts

```diff
@@ -0,0 +1,32 @@
+import { EchoSrv, EchoMeta, EchoEventType, EchoConsumer } from './types';
+import { PerformanceEvent } from './consumers/PerformanceConsumer';
+import { Echo } from './Echo';
+
+let instance: EchoSrv;
+
+export const getEcho = () => {
+  if (!instance) {
+    instance = new Echo({ debug: process.env.NODE_ENV === 'development' });
+    // @ts-ignore
+    window.Echo = instance;
```

Comment:
Just of curiosity why do we need the instance on `window`?

Reply:
Debug primarily, i'll leave a comment about that